Tools/environment_install/installROS-ubuntu.sh

```diff
@@ -0,0 +1,234 @@
+# Check to see if we have a ROS base kinda thingie
+hasBasePackage=false
+for package in "${packages[@]}"; do
+  if [[ $package == "ros-base" ]]; then
+     packages=" ros-${ROS_VERSION}-ros-base"
+     hasBasePackage=true
+     break
+  elif [[ $package == "desktop" ]]; then
+     packages=" ros-${ROS_VERSION}-desktop"    
+     hasBasePackage=true
+     break
+  elif [[ $package == "desktop-full" ]]; then
+     packages=" ros-${ROS_VERSION}-desktop-full"    
+     hasBasePackage=true
+     break
+  fi
+done
+if [ $hasBasePackage == false ] ; then
+   shouldInstallPackages
+   exit 1
+fi
+
+heading "${green}Adding Repositories and source lists${reset}"
+#Lets start instaling stuff
+sudo apt-add-repository universe
+sudo apt-add-repository multiverse
+sudo apt-add-repository restricted
+
+# Setup sources.lst
+sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list'
+# Setup keys
+sudo apt-key adv --keyserver 'hkp://keyserver.ubuntu.com:80' --recv-key C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654
+# If you experience issues connecting to the keyserver, you can try substituting hkp://pgp.mit.edu:80 or hkp://keyserver.ubuntu.com:80 in the previous command.
+# Installation
+
+heading "${green}Updating apt${reset}"
+sudo apt update
+
+heading "${green}Installing ROS${reset}"
+# Here we loop through any packages passed on the command line
+# Install packages ...
+for package in "${packages[@]}"; do
+  sudo apt install $package -y
+done
+
+# This is where you might start to modify the packages being installed, i.e.
+# sudo apt install ros-${ROS_VERSION}-{package_name}
+# sudo apt install -y setpriv
+# sudo apt install -y ros-${ROS_VERSION}-robot-upstart
+# sudo apt install -y ros-${ROS_VERSION}-navigation
+
+# install other needed packages
+sudo apt install build-essential cmake libglfw3-dev libglew-dev libeigen3-dev libjsoncpp-dev libtclap-dev -y
+
+#
+# To find available packages:
+# apt-cache search ros-melodic
+# 
+# Initialize rosdep
+
+heading "${green}Installing rosdep${reset}"
+
+sudo apt install python3-rosdep -y
+# Certificates are messed up on earlier version Jetson for some reason
+# Do not know if it is an issue with the Xavier, test by commenting out
+# sudo c_rehash /etc/ssl/certs
+# Initialize rosdep
+
+heading "${green}Initializaing rosdep${reset}"
+
+
+sudo rm /etc/ros/rosdep/sources.list.d/20-default.list && echo "OK" || echo "No 20-default.list file"
+
+sudo rosdep init
+# To find available packages, use:
+rosdep update
+# Use this to install dependencies of packages in a workspace
+# rosdep install --from-paths src --ignore-src --rosdistro=${ROS_DISTRO} -y
+# Environment Setup - Don't add /opt/ros/${ROS_VERSION}/setup.bash if it's already in bashrc
+heading "${green}Adding setup.bash, ROS_MASTER_URI and ROS_HOSTNAME to .bashrc ${reset}"
+grep -q -F "ROS_HOSTNAME=$HOSTNAME.local" ~/.bashrc || echo "ROS_HOSTNAME=$HOSTNAME.local" >> ~/.bashrc
```

Comment:
Those are needed if you want to communicate with an off board computer. You shouldn't need to plug a monitor and keyboard into your vehicles companion computer to use RVIZ for example. This makes the networking setup easier. And allows you to monitor ROS from a computer on a desk instead of in a vehicle.

Reply:
ArduPilot main repo isn't the best for CC specific config .
But main issue is that most won't use this and it could create issue specially if you don't have avahi/zeroconf enabled with .local setup as domain. Even on CC this is an assumption that cannot but trust 100% case.

Reply:
So what is your suggestion?

Reply:
I added a prompt to ask the user.

Reply:
not setting this. If somebody when to do remote ROS, it should look at the documentation how to do it.